File: web-forms/src/store/evaluation/endereco.js
import { defineStore } from 'pinia'
import { ref } from 'vue';

export const enderecoModel = {
    cep: null,
    estado: null,
    cidade: null,
    bairro: null,
    logradouro: null,
    numero: null,
    complemento: null,
    ponto_referencia: null
}

export const useEnderecoStore = defineStore('endereco', () => {
    
    const endereco = ref(JSON.parse(JSON.stringify(enderecoModel)))

    function reset() {
        endereco.value = JSON.parse(JSON.stringify(enderecoModel))
    }

    return {
        endereco,
        reset
    }
}, {
    persist: true
})

File: backend/internal/avaliation/domain/endereco.go
package domain

type Endereco struct {
	CEP             string  `json:"cep"`
	Estado          string  `json:"estado"`
	Cidade          string  `json:"cidade"`
	Bairro          string  `json:"bairro"`
	Logradouro      string  `json:"logradouro"`
	Numero          string  `json:"numero"`
	Complemento     *string `json:"complemento"`
	PontoReferencia *string `json:"ponto_referencia"`
}
